src/lib/provable/merkle-list.ts

```diff
@@ -335,19 +342,43 @@ class MerkleListIterator<T> implements MerkleListIteratorBase<T> {
     Object.assign(this, value);
   }
 
+  assertAtEnd() {
+    return this.currentHash.assertEquals(this.Constructor.emptyHash);
+  }
+
+  isAtStart() {
+    return this.currentHash.equals(this.hash);
+  }
```

Comment:
`assertAtEnd()` / `isAtEnd()` etc now don't align with `next()` / `previous()`. i.e. we call `isAtEnd()` inside `previous()` to check is we're done. it's the correct implementation but the wrong name: `isAtEnd()` should now be called `isAtStart()`

Reply:
we can push this to a follow-up PR though